bser.h

```diff
@@ -0,0 +1,104 @@
+#ifndef _LIBWATCHMAN_BSER_H_
+#define _LIBWATCHMAN_BSER_H_
+
+#include <stdint.h>
+
+#include "bser_private.h"
+
+/* ctors  - if 'fill' is NULL a new object will be allocated and returned,
+   otherwise, fill is initailized and returned */
+bser_t* bser_new_integer(uint64_t value, bser_t* fill);
+bser_t* bser_new_real(double value, bser_t* fill);
+bser_t* bser_new_true(bser_t* fill);
+bser_t* bser_new_false(bser_t* fill);
+bser_t* bser_new_null(bser_t* fill);
+bser_t* bser_new_string(const char* chars, size_t len, bser_t* fill);
```

Comment:
document here that string is *not* duplicated and that it is permitted to contain \0s.